reverse.js

```diff
@@ -20,12 +20,20 @@
    * and/or shortcut commands.
    */
   function normalizePath(d) {
+    // regular expression for detecting and replacing multiple decimal points
+    // in a continuous string
+    var regexp = /(\d*\.\d+)((\.\d*)+)/g;
+
     // preprocess "d" so that we have spaces between values
     d = d.replace(/,/g,' ')
          .replace(/([^eE])-/g,'$1 -')
          .replace(/\s*([achlmqstvzACHLMQSTVZ])\s*/g," $1 ")
          .replace(/\s+/g, ' ');
 
+    while (regexp.test(d)) {
+      d = d.replace(regexp, '$1 $2');
+    }
+
```

Comment:
no need for any ifs, we can just add this replacement as part of the four step replacement above. If the pattern matches, it'll replace as needed. If it doesn't, it's a noop.

Reply:
I tried to get it into the replacements above. Problem is it needs the loop. If we have something like `c.027.07.074.119.139.144` (taken from the example in #8) it would only replace the first occurrence producing  
`c.027 .07.074.119.139.144`

Reply:
You don't, because the replace function allows a regexp as first argument, and a _function_ as second argument for when you need more complex transforms:

```
var d = "c.027.07.074.119.139.144c.027.07.074.119.139.144";
d = d.replace(/,/g,' ')
         .replace(/([^eE])-/g,'$1 -')
         .replace(/\s*([achlmqstvzACHLMQSTVZ])\s*/g," $1 ")
         .replace(/\s+/g, ' ')
         .replace(/(\d*\.\d+)((\.\d*)+)/g, (a, b,c) => `${b}${c.split(`.`).join(` .`)}`);
```

yields ` c .027 .07 .074 .119 .139 .144 c .027 .07 .074 .119 .139 .144`.

Reply:
Oh, I didn't know about that. That's neat. I updated the code accordingly.